app/controllers/users_controller.rb

```diff
@@ -0,0 +1,49 @@
+class UsersController < ApplicationController
+  before_action :logged_in_user, except: %i(new show)
+  before_action :load_user, except: %i(index new)
+  before_action :correct_user, only: %i(edit update)
+
+  def index
+    @pagy, @users = pagy User.all, items: Settings.PAGE_10
+  end
+
+  def show
+    @user = current_user
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def edit; end
+
+  def update
+    if @user.update user_params
+      flash[:success] = I18n.t("model.user.update_success")
```

Comment:
create locales/model
```suggestion
      user:
         not_found: "User not found"
```